calculation/gmhazard_calc/gmhazard_calc/gms/GroundMotionDataset.py

```diff
@@ -459,21 +459,276 @@ def get_metadata_df(
+                        site_vs30,
+                    )
                 )
-            meta_df = pd.DataFrame.from_records(meta_data)
+            meta_df = pd.DataFrame.from_records(meta_data, index=selected_gms)
             meta_df.columns = ["mag", "rrup", "vs30"]
         else:
-            meta_df = pd.merge(self.source_metadata_df, site_source_df, how="left", left_on="fault", right_index=True)
+            # Need to filter based on actually available GM records
+            im_df = self.get_im_df(site_info, str(self.ims[0]))
+            source_df = self.source_metadata_df.loc[
+                np.isin(self.source_metadata_df.index.values, im_df.index.values)
+            ]
+
+            meta_df = pd.merge(
+                source_df,
+                site_source_df,
+                how="inner",
+                left_on="fault",
+                right_index=True,
+            )
             meta_df["vs30"] = site_vs30
             meta_df = meta_df[["fault", "mag", "rrup", "vs30"]]
 
         return meta_df
+
+
+class MixedGMDataset(GMDataset):
+    """
+    GM dataset that consists of both
+    observed and simulation records
+
+    For the case where GMS is run using
+    an parametric ensemble, but set of available
+    GM records to select from is supplemented with
+    simulations
+
+    Note: Currently only supports selecting from simulations
+    Todo: Complete this
+    """
+
+    def __init__(self, name):
+        super().__init__(name)
+
+        # Simulation
+        self.imdb_ffps = self._config["simulations"]["imdbs"]
+        self.simulation_dirs = self._config["simulations"]["waveforms_dirs"]
+        self.source_metadata_df = pd.read_csv(
+            self._config["simulations"]["source_metadata_ffp"], index_col=0
+        )
+
+        self.vs30_params_csv_ffp = self._config["simulations"]["vs30_params_csv_ffp"]
+        self.site_source_db_ffp = self._config["simulations"]["site_source_db_ffp"]
+        self.sources_dir = Path(self._config["simulations"]["sources_dir"])
+
+        self._ims = None
+
+    @property
+    def ims(self):
+        if self._ims is None:
+            for cur_imdb_ffp in self.imdb_ffps:
+                with dbs.IMDBNonParametric(cur_imdb_ffp) as imdb:
+                    cur_ims = [im for im in imdb.ims if IMType.has_value(im)]
+                    if self._ims is None:
+                        self._ims = set(cur_ims)
+                    else:
+                        self._ims.intersection_update(cur_ims)
+
+        self._ims = to_im_list(list(self._ims))
+
+        return self._ims
+
+    def get_waveforms(
+        self, gm_ids: Sequence[Any], site_info: site.SiteInfo, output_dir: str
+    ) -> List:
+        """
+        See GMDataset method for parameter specifications
+
+        Note: GM ids have the format: {sim_name}_{station_name}
+        """
+        return _get_sim_waveforms(self.simulation_dirs, gm_ids, output_dir)
```

Comment:
Is this meant to also pass through the site_info?

Reply:
Nope, the station name is in the GM id for simulations